### ISSUE TITLE
ServiceBusTrigger - Message Processing When Scaled-Out

### ISSUE THREAD

**Issue by phil000** [Azure/azure-functions-dotnet-worker#1756]
We have a service bus triggered function running on .NET 7, isolated mode.

We have scaled the app service plan to 2 nodes.

I have applied the following attribute to the function to ensure only a single instance of the listener is run regardless of how many nodes:
```
[Singleton(Mode = SingletonMode.Listener)]
```
However, the function logs indicated that the function was invoked twice still.

Is ```Microsoft.Azure.WebJobs.SingletonAttribute``` intended to work in isolated mode? 

What is the alternative to get a single instance function when the app service plan has > 1 node?

```
<PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.ServiceBus" Version="5.7.0" />
<PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" OutputItemType="Analyzer" />
<PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.10.0" />
<PackageReference Include="Microsoft.Azure.WebJobs.Core" Version="3.0.33" />
```

**Comment by SeanFeldman** [comment]
Is this possible duplicate of #274?

**Comment by phil000** [comment]
Possibly, although we'd really like some feedback or documentation on what the expected behavior of ServiceBusTrigger is expected to be when scaled-out over multiple servers, as it appears to have changed over time and we're really just guessing by observation. 

If we scale-out it _appears_ each message is processed once by one of the available servers. Basically as we'd expect, but when I logged this issue 6 months ago it didn't appear to be that way.

See: #2950

**Comment by SeanFeldman** [comment]
> See: #2950

Can you provide a link?

The title of this issue implies the focus is about `Singleton` but you're asking about the server scale-out.

**Comment by phil000** [comment]
https://github.com/Azure/azure-webjobs-sdk/issues/2950

Yes, if you scale out you want control over how many workers will receive and process a single message.

_Previously_ scaling-out _seemed_ to make every work receive and process the same messages. Aka a single message was process on each worker. Hence our need to apply a singleton to prevent this.

This behavior appeas to have changed (by observation) so a Singleton is perhaps no longer necessary and the current behavior works better, but it is undocumented behavior as far as I can see.

Basically in 6 months without a reply the issue has perhaps changed from requiring Singleton to work, to now requiring to know 
 how ServiceBusTrigger scales out.

**Comment by SeanFeldman** [comment]
Based on #274, it doesn't look like Isolated Worker SDK has support for single instance execution.
